packages/medusa/src/services/order.js

```diff
@@ -0,0 +1,447 @@
+        "Can't cancel an order with payment processed"
+      )
+    }
+
+    // cancel payment method
+
+    return this.orderModel_.updateOne(
+      {
+        _id: orderId,
+      },
+      {
+        $set: { status: "cancelled" },
+      }
+    )
+  }
+
+  /**
+   * Captures payment for an order.
+   * @param {string} orderId - id of order to capture payment for.
+   * @return {Promise} result of the update operation.
+   */
+  async capturePayment(orderId) {
+    const order = await this.retrieve(orderId)
+
+    if (order.payment_status !== "awaiting") {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Payment already captured"
+      )
+    }
+
+    const { provider_id, data } = order.payment_method
+    const paymentProvider = await this.paymentProviderService_.retrieveProvider(
+      provider_id
+    )
+
+    await paymentProvider.capturePayment(data)
+
+    return this.orderModel_.updateOne(
+      {
+        _id: orderId,
+      },
+      {
+        $set: { payment_status: "captured" },
+      }
+    )
+  }
+
+  /**
+   * Creates fulfillments for an order.
+   * In a situation where the order has more than one shipping method,
+   * we need to partition the order items, such that they can be sent
+   * to their respective fulfillment provider.
+   * @param {string} orderId - id of order to cancel.
+   * @return {Promise} result of the update operation.
+   */
+  async createFulfillment(orderId) {
+    const order = await this.retrieve(orderId)
+
+    if (order.fulfillment_status !== "not_fulfilled") {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Order is already fulfilled"
+      )
+    }
+
+    const { shipping_methods, items } = order
+
+    // if only one shipping method is attached to the order, we simply create
+    // the order for the single provider given in the method
+    if (shipping_methods.length === 1) {
+      const fulfillmentProviderId = shipping_methods[0].provider_id
+      const fulfillmentProvider = this.fulfillmentProviderService_.retrieveProvider(
+        fulfillmentProviderId
+      )
+
+      shipping_methods[0].items = items
+
+      await fulfillmentProvider.createOrder(shipping_methods[0].data, items)
+
+      return this.orderModel_.updateOne(
+        {
+          _id: orderId,
+        },
+        {
+          $set: { fulfillment_status: "fulfilled" },
+        }
+      )
+    }
+
+    // partition order items to their dedicated shipping method
+    order.shipping_methods = await this.partitionItems_(shipping_methods, items)
+
+    await Promise.all(
+      order.shipping_methods.map(method => {
+        const provider = this.fulfillmentProviderService_.retrieveProvider(
+          method.provider_id
+        )
+        provider.createOrder(method.data, method.items)
+      })
+    )
+
+    return this.orderModel_.updateOne(
+      {
+        _id: orderId,
+      },
+      {
+        $set: { fulfillment_status: "fulfilled" },
```

Comment:
TLDR: Check if we need to update `status` as well
For more, see my comment about statuses in `models/order.js`.